pvlib/ivtools.py

```diff
@@ -262,6 +268,173 @@ def fit_sde_sandia(voltage, current, v_oc=None, i_sc=None, v_mp_i_mp=None,
                                      v_oc)
 
 
+def fit_sdm_desoto(celltype, v_mp, i_mp, v_oc, i_sc, alpha_sc, beta_voc,
+                   cells_in_series=60, temp_ref=25):
+    """
+    Calculates the five parameters for the single diode equation using
+    the De Soto et al. procedure described in [1]. This procedure has the
+    advantage of using common specifications given by manufacturers in the
+    datasheets of PV modules.
+    The six values returned by this function can be used by
+    pvsystem.calcparams_desoto to calculate the values at different
+    irradiance and cell temperature.
+
+    Parameters
+    ----------
+    celltype : str, case insensitive
+        Value is one of 'monosi', 'multisi', 'polysi' or'gaas'.
+        Others like 'cis', 'cigs', 'cdte', 'amorphous' are not implemented yet
+    v_mp : numeric
+        Module voltage at the maximum-power point at std conditions in V.
+    i_mp : numeric
+        Module current at the maximum-power point at std conditions in A.
+    v_oc : numeric
+        Open-circuit voltage at std conditions in V.
+    i_sc : numeric
+        Short-circuit current at std conditions in A.
+    alpha_sc : numeric
+        The short-circuit current (i_sc) temperature coefficient of the
+        module in units of %/K. It is converted in A/K for the computing
+        process.
+    beta_voc : numeric
+        The open-circuit voltage (v_oc) temperature coefficient of the
+        module in units of %/K. It is converted in V/K for the computing
+        process.
+    cells_in_series : numeric, default 60
+        Number of cell in the module.
+        Optional input, but helps to insure the convergence of the computing.
```

Comment:
Since you know `celltype`, you might try making the default value here `None`, in which case you estimate `cells_in_series` by dividing `v_oc` by a "nominal" `v_oc` for a single cell of the given `celltype`.

Reply:
I think it is safe to assume that `cells_in_series` is known. Should be required input.

Reply:
In this case I think it’s better to require explicitness here, and not provide a default value.